api/src/main/java/org/pmiops/workbench/api/DiskController.java

```diff
@@ -0,0 +1,124 @@
+import org.pmiops.workbench.leonardo.model.LeonardoListPersistentDiskResponse;
+import org.pmiops.workbench.model.Disk;
+import org.pmiops.workbench.model.EmptyResponse;
+import org.pmiops.workbench.notebooks.LeonardoNotebooksClient;
+import org.pmiops.workbench.utils.mappers.LeonardoMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class DiskController implements DiskApiDelegate {
+  private static final Logger log = Logger.getLogger(DiskController.class.getName());
+  private final LeonardoNotebooksClient leonardoNotebooksClient;
+  private final WorkspaceDao workspaceDao;
+  private final LeonardoMapper leonardoMapper;
+  private final Provider<DbUser> userProvider;
+  // timestamp for the date was created in ISO 8601 format
+  private static final String diskCreatedDateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'";
+
+  @Autowired
+  public DiskController(
+      LeonardoNotebooksClient leonardoNotebooksClient,
+      WorkspaceDao workspaceDao,
+      Provider<DbUser> userProvider,
+      LeonardoMapper leonardoMapper) {
+    this.leonardoNotebooksClient = leonardoNotebooksClient;
+    this.workspaceDao = workspaceDao;
+    this.userProvider = userProvider;
+    this.leonardoMapper = leonardoMapper;
+  }
+
+  @Override
+  public ResponseEntity<Disk> getDisk(String workspaceNamespace) {
+    String googleProject = lookupWorkspace(workspaceNamespace).getGoogleProject();
+    String pdNamePrefix = userProvider.get().getUserPDNamePrefix();
+    SimpleDateFormat dateFormatter = new SimpleDateFormat(diskCreatedDateFormat);
+    AtomicInteger activePDCnt = new AtomicInteger();
+    AtomicInteger failedPDCnt = new AtomicInteger();
+    Stream<LeonardoListPersistentDiskResponse> responseStream =
+        leonardoNotebooksClient.listPersistentDiskByProject(googleProject, false).stream()
+            .filter(r -> r.getName().startsWith(pdNamePrefix))
+            .peek(
+                r -> {
+                  if (r.getStatus() != null) {
+                    switch (r.getStatus()) {
+                      case READY:
+                        activePDCnt.addAndGet(1);
+                        break;
+                      case FAILED:
+                        failedPDCnt.addAndGet(1);
+                        break;
+                      default:
+                        break;
+                    }
+                  }
+                }) // collect disk status
+            .sorted(
+                (r1, r2) -> {
+                  try {
+                    return dateFormatter
+                        .parse(r2.getAuditInfo().getCreatedDate())
+                        .compareTo(dateFormatter.parse(r1.getAuditInfo().getCreatedDate()));
+                  } catch (ParseException e) {
+                    log.warning(String.format("Observed PDs CreatedDate parsing error: %s", e));
```

Comment:
Feels this is not correct logic: A ParseException is throwed, but we just return 0. The 0 means NotFoundException in your line 82 below.
 I think   ParseException should mapped to 500 error (ServerErrorException), not 404